File: content/examples/other/leetcode/problems/find-minimum-in-rotated-sorted-array/example-1/solution.js
function findMinLinear(nums) {
  let best = Infinity;
  for (const x of nums) best = Math.min(best, x);
  return best;
}

if (require.main === module) {
  console.log(findMinLinear([3, 4, 5, 1, 2]));
}

module.exports = { findMinLinear };

File: content/examples/other/leetcode/problems/find-minimum-in-rotated-sorted-array/example-1/Solution.java
public class Solution {
  static int findMinLinear(int[] nums) {
    int best = Integer.MAX_VALUE;
    for (int x : nums) best = Math.min(best, x);
    return best;
  }

  public static void main(String[] args) {
    System.out.println(findMinLinear(new int[] {3, 4, 5, 1, 2}));
  }
}
